datasets/cord19/cord19.py

```diff
@@ -0,0 +1,207 @@
+  D. Raymond and Daniel S. Weld and Oren Etzioni and Sebastian Kohlmeier},
+  journal={ArXiv},
+  year={2020}
+}
+"""
+
+# You can copy an official description
+_DESCRIPTION = """\
+The Covid-19 Open Research Dataset (CORD-19) is a growing resource of scientific papers on Covid-19 and related
+historical coronavirus research. CORD-19 is designed to facilitate the development of text mining and information
+retrieval systems over its rich collection of metadata and structured full text papers. Since its release, CORD-19
+has been downloaded over 75K times and has served as the basis of many Covid-19 text mining and discovery systems.
+
+The dataset itself isn't defining a specific task, but there is a Kaggle challenge that define 17 open research
+questions to be solved with the dataset: https://www.kaggle.com/allen-institute-for-ai/CORD-19-research-challenge/tasks
+"""
+
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URL = "https://ai2-semanticscholar-cord-19.s3-us-west-2.amazonaws.com/historical_releases/cord-19_2020-11-29.tar.gz"
+
+
+class Cord19(datasets.GeneratorBasedBuilder):
+    """The Covid-19 Open Research Dataset (CORD-19) is a growing resource of scientific papers on Covid-19."""
+
+    VERSION = datasets.Version("0.0.1")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="metadata",
+            description="The set of documents but loading some metadata like title and " "abstract for each article.",
+        ),
+        datasets.BuilderConfig(
+            name="fulltext",
+            description="The set of documents loading some metadata like title and "
+            "abstract and full text for each article.",
+        ),
+        datasets.BuilderConfig(
+            name="embeddings",
+            description="The set of documents loading some metadata like title and "
+            "abstract and document embeddings for each article.",
+        ),
+    ]
+
+    def _info(self):
+        # default metadata only
+        features_dict = {
+            "cord_uid": datasets.Value("string"),
+            "sha": datasets.Value("string"),
+            "source_x": datasets.Value("string"),
+            "title": datasets.Value("string"),
+            "doi": datasets.Value("string"),
+            "abstract": datasets.Value("string"),
+            "publish_time": datasets.Value("string"),
+            "authors": datasets.Value("string"),
+            "journal": datasets.Value("string"),
+            "url": datasets.Value("string"),
+        }
+
+        if "fulltext" in self.config.name:
+            # adding full_text
+            features_dict["fulltext"] = datasets.Value("string")
+
+        if "embeddings" in self.config.name:
+            # adding embeddings
+            features_dict["doc_embeddings"] = datasets.Sequence(datasets.Value("float64"))
```

Comment:
Should it be float64 or float32 ?

Reply:
The values seems to have a high precision and necessitate float64
```
ug7v899j,-2.939983606338501,-6.312200546264648,-1.0459030866622925,5.164162635803223 [...]
```

Reply:
Ok :)